batan-server/app/state/game.ts

```diff
@@ -168,17 +171,54 @@ export default {
         this.nextTurn(game_id, game.turn_num, callback);
       }, 180000);
 
-      let turnStartData = {
-        turn_type: "normal",
-        current_player: this.whosTurn(game_id),
-        turn_over_time: end_turn_time.toISOString()
+      game.end_turn_time = end_turn_time.toISOString();
+
+      let turnStartData = this.get_full_game_info(game_id);
+
+      callback(turnStartData);
+    }
+  },
+  get_full_game_info(game_id){
+    let game = this.games.get(game_id);
+
+    let scores = []
+    for (let player_num in game.order) {
+      let player = game.gameObj.players[player_num];
+      let dc = player.developmentCards;
+      let p_scores = {
+        name: player.name,
+        victoryPoints: player.victoryPoints,
+        developmentCards: dc.knight + dc.victoryPointCard + dc.roadBuilder + dc.yearOfPlenty + dc.monopoly,
+        settlementsPlayed: player.settlementsPlayed,
+        citiesPlayed: player.citiesPlayed,
+        roadsPlayed: player.roadsPlayed,
+        longestRoad: player.longestRoad,
+        armies: player.armies,
+        vpDevCardsPlayed: player.vpDevCardsPlayed
       }
+      scores.push(p_scores);
+    }
 
-      if (game.turn_num < game.players.size*2) {
-        turnStartData.turn_type = "init";
+    let dc = game.gameObj.bank.developmentCards;
+    let turnStartData = {
+      game_id: game_id,
+      turn: {
+        type: "normal",
+        player: this.whosTurn(game_id),
+        over_at: game.end_turn_time
+      },
+      board: JSON.stringify(game.gameObj.board, game.gameObj.replacer),
+      score: scores,
+      bank: {
+        resources: JSON.stringify(game.gameObj.bank.resources, game.gameObj.replacer),
+        developmentCards: dc.knight + dc.victoryPointCard + dc.roadBuilder + dc.yearOfPlenty + dc.monopoly
       }
-      callback(turnStartData);
     }
+
+    if (game.turn_num < game.players.size*2) {
```

Comment:
I'm not sure I understand the * 2 part. Could you explain?

Reply:
The first 'turns' in the game we place 1 settlement + 1 road in forward order, then reverse - it's just checking if it's in one of those first placement turns

Reply:
gotcha. Go for merge